### PR TITLE
Expand Test Suite

### PR DESCRIPTION
**Pull request by jeremylt**
This PR adds some minor tests for important CeedVector error cases and Identity QFunctions with size>1. This also fixes a double free issue discovered due to these tests.

Partially fixes Issue #357
Fixes Issue #354

### CODE REVIEW

File: tests/tap.sh [hunk 1]
```diff
@@ -74,9 +74,9 @@ for ((i=0;i<${#backends[@]};++i)); do
         continue
     fi
 
-    # grep to pass test t103, t104, t105, t106, t107 on error
+    # grep to pass test t11* on error
```
Comment (jeremylt) [inline tests/tap.sh]: Reordered tests to make `tap.sh` cleaner.

File: tests/output/t302-basis-f.out [hunk 2]
```diff
@@ -1,48 +1,11 @@
...
-   qweight1d:	  0.16666667	  0.83333333	  0.83333333	  0.16666667
-    interp1d[0]:	  1.00000000	  0.00000000	  0.00000000	  0.00000000
-    interp1d[1]:	  0.00000000	  1.00000000	  0.00000000	  0.00000000
-    interp1d[2]:	  0.00000000	  0.00000000	  1.00000000	  0.00000000
-    interp1d[3]:	  0.00000000	  0.00000000	  0.00000000	  1.00000000
-      grad1d[0]:	 -3.00000000	  4.04508497	 -1.54508497	  0.50000000
-      grad1d[1]:	 -0.80901699	  0.00000000	  1.11803399	 -0.30901699
-      grad1d[2]:	  0.30901699	 -1.11803399	  0.00000000	  0.80901699
-      grad1d[3]:	 -0.50000000	  1.54508497	 -4.04508497	  3.00000000
 collograd[0]: -3.00000000  4.04508497 -1.54508497  0.50000000
```
Comment (valeriabarra) [inline tests/output/t302-basis-f.out]: Are the `out` files supposed to be pushed?

Comment (jeremylt) [reply tests/output/t302-basis-f.out]: Yes, the out files are the comparison files for the tests that intentionally write output.

Comment (valeriabarra) [reply tests/output/t302-basis-f.out]: Yes, thanks. I had just misread the path. The question was meant to ask about being "pushed here" and I typed too fast

**Pull request merged**